Packages/com.unity.inputsystem/InputSystem/NativeInputRuntime.cs

```diff
@@ -198,14 +215,43 @@ private void OnFocusChanged(bool focus)
         public bool isInPlayMode => EditorApplication.isPlaying;
         public bool isPaused => EditorApplication.isPaused;
 
+        private Action<PlayModeStateChange> m_OnPlayModeChanged;
+        private Action m_OnProjectChanged;
+
+        private void OnPlayModeStateChanged(PlayModeStateChange value)
+        {
+            m_OnPlayModeChanged(value);
+        }
+
+        private void OnProjectChanged()
+        {
+            m_OnProjectChanged();
+        }
+
         public Action<PlayModeStateChange> onPlayModeChanged
         {
-            set => EditorApplication.playModeStateChanged += value;
+            get => m_OnPlayModeChanged;
+            set
+            {
+                if (value == null)
+                    EditorApplication.playModeStateChanged -= OnPlayModeStateChanged;
+                else if (m_FocusChangedMethod == null)
+                    EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
+                m_OnPlayModeChanged = value;
+            }
         }
 
         public Action onProjectChange
         {
-            set => EditorApplication.projectChanged += value;
+            get => m_OnProjectChanged;
+            set
+            {
+                if (value == null)
+                    EditorApplication.projectChanged -= OnProjectChanged;
+                else if (m_FocusChangedMethod == null)
```

Comment:
What's the m_FocusChangedMethod check here and in the setter above about?

Reply:
oops, copy-paste mistake - will fix